server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -323,6 +325,27 @@ public void testValidator() {
         assertTrue(FooBarValidator.invokedWithDependencies);
     }
 
+    public void testRegexValidator() throws Exception {
+        // A regex that matches one or more digits
+        String expectedRegex = "\\d+";
+        Pattern expectedPattern = Pattern.compile(expectedRegex);
+        RegexValidator regexValidator = new RegexValidator(expectedRegex);
+
+        // Test that the pattern is correctly initialized
+        assertNotNull(expectedPattern);
+        assertNotNull(regexValidator.getPattern());
+        assertEquals(expectedPattern.toString(), regexValidator.getPattern().toString());
+
```

Comment:
You need to add a test of the validator itself.  

Since you've given it a test pattern of at least one digit, any string with just digits should throw an exception and any other string should not.

```java
assertThrows(IllegalArgumentException.class, () -> regexValidator.validate("123"));
assertDoesNotThrow(() -> regexValidator.validate("abc"));
```

Note the pattern you give is what fails the validation so if you want to ensure it's just digits you could use the `\\D+` pattern to fail if it's anything other than digits and reverse the above two tests.

Repeat these tests below after writing and reading the parser.

Reply:
Try to Use `try {
    regexValidator.validate("abc");
} catch (IllegalArgumentException e) {
    fail("Expected validate() to not throw an exception, but it threw " + e);
}
` instead of 
`assertThrows(IllegalArgumentException.class, () -> regexValidator.validate("123"));
assertDoesNotThrow(() -> regexValidator.validate("abc"));` 
because the error message shows 
`> Task :client:test:loggerUsageCheck
        assertDoesNotThrow(() -> regexValidator.validate("abc"));
        ^
  symbol:   method assertDoesNotThrow(()->regexV[...]abc"))
  location: class SettingTests
/home/runner/work/OpenSearch/OpenSearch/server/src/test/java/org/opensearch/common/settings/SettingTests.java:348: error: cannot find symbol
                assertDoesNotThrow(() -> regexValidator.validate("abc"));
                ^
  symbol:   method assertDoesNotThrow(()->regexV[...]abc"))
  location: class SettingTests`
after doing some research I find out that `assertDoesNotThrow()` is a JUnit 5 feature that was introduced in version 5.2.0, so if we are using an earlier version of JUnit 5, the method will not be available. But I am not able to find how to check the Junit version, so I wrap the call to validate() in a try-catch block and use fail() to signal that an exception was thrown.

Reply:
Ah, correct. I knew they added `assertThrows()` in JUnit 4.13 (which is used here) but didn't know the opposite didn't work.  I think your try/catch and fail in the catch block is a good option instead.

Reply:
@mloufra you added the "not throws" case but didn't add the assertion for the expected exception.

Reply:
ACK